process/heartbeatProcessor.go

```diff
@@ -44,6 +51,56 @@ func NewHeartbeatProcessor(
 	return hbp, nil
 }
 
+// IsOldStorageForToken returns true if the token is stored in the old fashion
+func (hbp *HeartbeatProcessor) IsOldStorageForToken(tokenID string, nonce uint64) (bool, error) {
+	observers, err := hbp.proc.GetAllObservers()
+	if err != nil {
+		return false, err
+	}
+
+	tokenStorageKey := computeTokenStorageKey(tokenID, nonce)
+	for _, observer := range observers {
+		if observer.ShardId == core.MetachainShardId {
+			continue
+		}
+
+		apiResponse := data.AccountKeyValueResponse{}
+		apiPath := AddressPath + systemAccountAddress + "/key/" + tokenStorageKey
+		respCode, err := hbp.proc.CallGetRestEndPoint(observer.Address, apiPath, &apiResponse)
+		if err == nil || respCode == http.StatusBadRequest || respCode == http.StatusInternalServerError {
+			log.Info("account value for key request",
+				"address", systemAccountAddress,
+				"shard ID", observer.ShardId,
+				"observer", observer.Address,
+				"http code", respCode)
+			if apiResponse.Error != "" {
+				return false, errors.New(apiResponse.Error)
+			}
+
+			log.Info("load token from system account", "token", tokenID, "nonce", nonce, "shard ID", observer.ShardId, "value", apiResponse.Data.Value)
+			if len(apiResponse.Data.Value) > 0 {
+				return false, nil
+			}
+		}
+	}
+
+	return true, nil
```

Comment:
If there's an `err` on each observer, we return `true, nil`. Seems all right - please confirm.

Reply:
fixed, if any of the node errors, the function will return error